apps/cloudbroker/base/cloudapi__machines/methodclass/cloudapi_machines.py

```diff
@@ -113,12 +114,19 @@ def start(self, machineId, diskId=None, **kwargs):
             disk = self.models.disk.get(diskId)
             if disk.type != 'C':
                 raise exceptions.BadRequest("diskId is not of type CD-ROM")
+            if disk.status in resourcestatus.Image.INVALID_STATES:
+                raise exceptions.BadRequest("Rescue disk with id {} is not available".format(diskId))
             cloudspace = self.models.cloudspace.get(machine.cloudspaceId)
             if disk.accountId and disk.accountId != cloudspace.accountId:
                 raise exceptions.BadRequest("Rescue disk with id {} belongs to another account".format(diskId))
             volume = j.apps.cloudapi.disks.getStorageVolume(disk, provider, node)
             node.extra['volumes'].append(volume)
             node.extra['bootdev'] = 'cdrom'
+            tags.tagSet('cdrom', disk.id)
```

Comment:
tags need to be string I beleive sthi will crash you need to do str(disk.id)

Reply:
No it doesn't have to be